wolfSSL/src/chapter02.md

```diff
@@ -394,301 +394,1177 @@ Removes code for OAEP padding.
+
+#### WOLFSSL_QUIC_H
+
+Define for wolSSL quic API.
+
+#### WOLFSSL_QUIC_MAX_RECORD_CAPACITY
+
+Defines max quic capacity as 1024*1024 -- 1 MB.
+
+#### WOLFSSL_RENESAS_FSPSM_TLS
+
+Not yet supported TLS related capabilities.
+
+#### WOLFSSL_RENESAS_TSIP_TLS
+
+This is for disabling only the TSIP TLS-linked common key encryption method.
+
+#### WOLFSSL_SM2
+
+Define to use SM ciphers.
+
+#### WOLFSSL_SM3
+
+Define to use SM ciphers.
+
+#### WOLFSSL_SM4
+
+Define to use SM ciphers.
+
+#### WOLFSSL_SM4_CBC
+
+SM setting for SM4 CBC.
+
+#### WOLFSSL_SM4_CCM
+
+SM settings for SM4 CCM.
+
+#### WOLFSSL_SM4_GCM
+
+SM settings for SM4 GCM.
+
+#### WOLFSSL_SNIFFER_CHAIN_INPUT
+
+Used in conjunction with build option ./configure --enable-sniffer.
+
+#### XSLEEP_MS
+
+Can be called to make the session wait for a specified number of milliseconds till it expires.
+
+#### XSNPRINTF
+
+Snprintf is used in asn.c for GetTimeString, PKCS7 test, and when debugging is turned on.
+
+#### DEFAULT_TIMEOUT_SEC
+
+Used to define a default timeout in seconds.
+
+#### HAVE_CSHARP
+
+Turns on configuration options needed for C# wrapper.
+
+#### HAVE_CURVE25519
+
+Turns on the use of curve25519 algorithm.
+
+#### HAVE_ED25519
+
+Turns on use of the ed25519 algorithm.
+
+#### WOLFSSL_DH_CONST
+
+Turns off use of floating point values when performing Diffie Hellman operations and uses tables for `XPOW()` and `XLOG()`. Removes dependency on external math library.
+
+#### WOLFSSL_TRUST_PEER_CERT
+
+Turns on the use of trusted peer certificates. This allows for loading in a peer certificate to match with a connection rather than using a CA. When turned on if a trusted peer certificate is matched than the peer cert chain is not loaded and the peer is considered verified. Using CAs is preferred.
+
+#### WOLFSSL_STATIC_MEMORY
+
+Turns on the use of static memory buffers and functions. This allows for using static memory instead of dynamic.
+
+#### WOLFSSL_STATIC_MEMORY_LEAN
+
+It requires WOLFSSL_STATIC_MEMORY to be defined. It uses smaller type sizes for structs requiring memory pool sizes of less than 65k and limits features available, like IO buffers, to reduce footprint size.
+
+#### WOLFSSL_SESSION_EXPORT
+
+Turns on the use of DTLS session export and import. This allows for serializing and sending/receiving the current state of a DTLS session.
+
+#### WOLFSSL_ARMASM
+
+Turns on the use of ARMv8 hardware acceleration.
+
+#### WC_RSA_NONBLOCK
+
+Turns on fast math RSA non-blocking support for splitting RSA operations into smaller chunks of work. Feature is enabled by calling [`wc_RsaSetNonBlock()`](group__RSA.md#function-wc_rsasetnonblock) and checking for `FP_WOULDBLOCK` return code.
+
+#### WC_RNG_TYPE_DEFINED
+
+Used as a guard on redeclaration.
+
+#### WOLFSSL_DH_TYPE_DEFINED
+
+Used as redeclaration gaurd for DH.
```

Comment:
spelling -- "guard" as above, not "gaurd".  There's a bunch of other misspelled "gaurd"s below too.